nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/auth/UserResolver.java

```diff
@@ -48,15 +48,13 @@ public OpenIDConnectProvider getProvider() {
 
     protected  DocumentModel createNuxeoUser(String nuxeoLogin) {
         DocumentModel userDoc;
-
+        UserManager userManager = Framework.getService(UserManager.class);
+        DocumentModel user = userManager.getBareUserModel();
+        user.setPropertyValue(userManager.getUserIdField(), nuxeoLogin);
         try {
-            UserManager userManager = Framework.getService(UserManager.class);
-
-            userDoc = userManager.getBareUserModel();
-            userDoc.setPropertyValue(userManager.getUserIdField(), nuxeoLogin);
-
-            userDoc = userManager.createUser(userDoc);
-
+            userDoc = Framework.doPrivileged(() -> {
+                return userManager.createUser(user);
+            });
```

Comment:
Sonar would tell you to "Remove useless curly braces around statement and then remove useless return keyword":
```
userDoc = Framework.doPrivileged(() -> userManager.createUser(user));
```

Reply:
agreed, but I prefer to keep the same code as in the master branch - this commit is a simple backport of an existing fix